pkg/operator/ceph/object/spec.go

```diff
@@ -406,35 +440,77 @@ func configureReadinessProbe(container *v1.Container, healthCheck cephv1.ObjectH
 	}
 }
 
-func (c *clusterConfig) defaultLivenessProbe() *v1.Probe {
-	return &v1.Probe{
+func (c *clusterConfig) defaultLivenessProbe() (*v1.Probe, error) {
```

Comment:
Why not just not set the LivenessProbe on the spec? If preferred to keep this method for clarity, maybe rename it to something like `noLivenessProbe()`? And no need to return an error.

Reply:
I see benefit to changing this to `noLivenessProbe()` for clarity. But it returns an error so its function signature matches the other probe functions. That allows it to be tested for all the same unit test cases more easily.

Reply:
Actually, there's not reason to test the `defaultLivenessProbe()` function anyway since it just returns nil and should always unless something changes. 🤦‍♀️